### PR TITLE
Demonstrate issue #3319

### PR DESCRIPTION
**Pull request by ofekd**
Please see issue #3319

### CODE REVIEW

**Comment by pubkey** [comment]
I think this is solved in the latest version.
Can you rebase from master branch?

**Comment by ofekd** [comment]
@pubkey done, `getLastPullDocument` isn't resolving instead of returning `null`

**Comment by pubkey** [comment]
Is this solved in the latest release?

**Comment by ofekd** [comment]
@pubkey apologies, I was unavailable. `getLastPullDocument` still isn't resolving for pouch. For loki (amazing addition, I will be migrating), it errors out:

```

  1) replication-graphql.test.js
       node and browser
         crawling-checkpoint
           .getLastPullDocument()
             should delete the doc after db.remove():
     Error: BroadcastChannel.postMessage(): Cannot post message after channel has closed
      at BroadcastChannel.postMessage (node_modules/broadcast-channel/dist/es5node/broadcast-channel.js:107:13)
      at RxStorageKeyObjectInstanceLoki._callee3$ (src/plugins/lokijs/rx-storage-key-object-instance-loki.ts:188:26)
      at tryCatch (node_modules/regenerator-runtime/runtime.js:63:40)
      at Generator.invoke [as _invoke] (node_modules/regenerator-runtime/runtime.js:294:22)
      at Generator.next (node_modules/regenerator-runtime/runtime.js:119:21)
      at asyncGeneratorStep (node_modules/@babel/runtime/helpers/asyncToGenerator.js:3:24)
      at _next (node_modules/@babel/runtime/helpers/asyncToGenerator.js:25:9)
      at /home/ofek/rxdb/node_modules/@babel/runtime/helpers/asyncToGenerator.js:32:7
      at new Promise (<anonymous>)
      at RxStorageKeyObjectInstanceLoki.<anonymous> (node_modules/@babel/runtime/helpers/asyncToGenerator.js:21:12)
      at RxStorageKeyObjectInstanceLoki.requestRemoteInstance (dist/lib/plugins/lokijs/rx-storage-key-object-instance-loki.js:255:37)
      at RxStorageKeyObjectInstanceLoki._callee5$ (src/plugins/lokijs/rx-storage-key-object-instance-loki.ts:325:25)
      at tryCatch (node_modules/regenerator-runtime/runtime.js:63:40)
      at Generator.invoke [as _invoke] (node_modules/regenerator-runtime/runtime.js:294:22)
      at Generator.next (node_modules/regenerator-runtime/runtime.js:119:21)
      at asyncGeneratorStep (node_modules/@babel/runtime/helpers/asyncToGenerator.js:3:24)
      at _next (node_modules/@babel/runtime/helpers/asyncToGenerator.js:25:9)
```

I have pushed a rebased version

**Comment by pubkey** [comment]
There is a similar error when running the current tests many times: https://github.com/pubkey/rxdb/runs/4384515025?check_suite_focus=true
So this is very likely a RxDB bug

**Comment by ofekd** [comment]
It looks like in this specific test it happens because `db.remove()` closes `broadcastChannel`, and then it can't be used for `getLastPullDocument`. There's probably something similar happening in the pouch case

File: test/unit/replication-graphql.test.ts [hunk 1]
```diff
@@ -681,6 +681,22 @@ describe('replication-graphql.test.js', () => {
...
+                    const c = await humansCollection.createHumanWithTimestamp(
+                        1
+                    );
+                    const doc = await c.findOne().exec(true);
+                    let docData = doc.toJSON(true);
+                    docData = clone(docData); // clone to make it mutateable
+                    (docData as any).name = 'foobar';
+
+                    await setLastPullDocument(c, endpointHash, docData);
+                    await c.database.remove();
```
Comment (pubkey) [inline test/unit/replication-graphql.test.ts:694]: Wait. So you remove the database and then query it again? I am not sure if this is supposed to work.

Comment (ofekd) [reply test/unit/replication-graphql.test.ts:694]: Originally that was to show the local documents persist DB removal. Now that they are removed, the question is only if `getLastPullDocument` should fail or return null

Not sure why other `loki` test errors, though

Comment (pubkey) [reply test/unit/replication-graphql.test.ts:694]: Running any operation on a removed collection/database should fail.
To query it again, a new RxDatabase must be created.

**Comment by pubkey** [comment]
Running any operation on a removed collection should fail

**Pull request closed without merging**